Pycuda failed to compile during installation

**Describe the bug**
Not able to install pycuda.
Error
```
In file included from src/wrapper/wrap_curand.cpp:2:
src/cpp/curand.hpp: In function ‘void pycuda::curandom::py_curand_get_direction_vectors(curandDirectionVectorSet_t, boost::python::api::object, int)’:
src/cpp/curand.hpp:55:26: warning: unused variable ‘len’ [-Wunused-variable]
   55 |     PYCUDA_BUFFER_SIZE_T len = buf_wrapper.m_buf.len;
      |                          ^~~
In file included from src/wrapper/wrap_curand.cpp:2:
src/cpp/curand.hpp: In function ‘void pycuda::curandom::py_curand_get_scramble_constants32(boost::python::api::object, int)’:
src/cpp/curand.hpp:96:26: warning: unused variable ‘len’ [-Wunused-variable]
   96 |     PYCUDA_BUFFER_SIZE_T len = buf_wrapper.m_buf.len;
      |                          ^~~
src/cpp/curand.hpp: In function ‘void pycuda::curandom::py_curand_get_scramble_constants64(boost::python::api::object, int)’:
src/cpp/curand.hpp:118:26: warning: unused variable ‘len’ [-Wunused-variable]
  118 |     PYCUDA_BUFFER_SIZE_T len = buf_wrapper.m_buf.len;
      |                          ^~~
In file included from /nix/store/9jl7rqxx2x5y8x2nv5h6zi93x6aw97v7-python3.7-numpy-1.18.0/lib/python3.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:21,
                 from /nix/store/9jl7rqxx2x5y8x2nv5h6zi93x6aw97v7-python3.7-numpy-1.18.0/lib/python3.7/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                 from src/wrapper/tools.hpp:10,
                 from src/wrapper/wrap_curand.cpp:4:
/nix/store/9jl7rqxx2x5y8x2nv5h6zi93x6aw97v7-python3.7-numpy-1.18.0/lib/python3.7/site-packages/numpy/core/include/numpy/__multiarray_api.h: At global scope:
/nix/store/9jl7rqxx2x5y8x2nv5h6zi93x6aw97v7-python3.7-numpy-1.18.0/lib/python3.7/site-packages/numpy/core/include/numpy/__multiarray_api.h:1463:1: warning: ‘int _import_array()’ defined but not used [-Wunused-function]
 1463 | _import_array(void)
      | ^~~~~~~~~~~~~
g++ -pthread -shared -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.2.11/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bzip2-1.0.6.0.1/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.2.8/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xz-5.2.4/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libffi-3.3/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.18.1/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-sqlite-3.30.1/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-6.3p08/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.1-20190112/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-1.1.1d/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.2.11/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bzip2-1.0.6.0.1/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.2.8/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xz-5.2.4/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libffi-3.3/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.18.1/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-sqlite-3.30.1/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-6.3p08/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.1-20190112/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-1.1.1d/lib build/temp.linux-x86_64-3.7/src/cpp/cuda.o build/temp.linux-x86_64-3.7/src/cpp/bitlog.o build/temp.linux-x86_64-3.7/src/wrapper/wrap_cudadrv.o build/temp.linux-x86_64-3.7/src/wrapper/mempool.o build/temp.linux-x86_64-3.7/src/wrapper/wrap_curand.o -L/nix/store/5rbsf0cl5zvfaswmfqnqmkzxmzhh388v-boost-1.69.0/lib -L/nix/store/bzvpdbzdy3a1qba8f9ynrzyz811c10pp-python3-3.7.6/lib -lboost_python3 -lboost_thread -lcuda -lcurand -lpython3.7m -o build/lib.linux-x86_64-3.7/pycuda/_driver.cpython-37m-x86_64-linux-gnu.so
/nix/store/5nxyf0yn836j0acm82cz3g50d4glk5df-binutils-2.31.1/bin/ld: cannot find -lboost_python3
/nix/store/5nxyf0yn836j0acm82cz3g50d4glk5df-binutils-2.31.1/bin/ld: cannot find -lcuda
collect2: error: ld returned 1 exit status
error: command 'g++' failed with exit status 1
builder for '/nix/store/qbrbaa8p0njjvhpig2hrn2amzz9mq50p-python3.7-pycuda-2019.1.2.drv' failed with exit code 1
cannot build derivation '/nix/store/spd6rlsvjfxj6hyc0f0fmm6z696z0cg6-python3-3.7.6-env.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/0br4y01y98mnmv96k49dddn9ipwc8n8q-user-environment.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/bm8xpga8d95yxricrywi4rfsk9cjal2p-etc.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/plmpz16112qg1r3yq37j0857bq2kncsi-nixos-system-lenovo-nixos-19.09.1840.f7d050ed4e3.drv': 1 dependencies couldn't be built
error: build of '/nix/store/plmpz16112qg1r3yq37j0857bq2kncsi-nixos-system-lenovo-nixos-19.09.1840.f7d050ed4e3.drv' failed
```

**To Reproduce**
Install pycuda

**Expected behavior**
install pycuda with dependencies

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.10, NixOS, 19.09.1821.9f453eb97ff (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - channels(root): `"home-manager-19.09, nixos-19.09.1840.f7d050ed4e3, unstable-20.03pre208985.aa561c6fb4f"`
 - channels(mudrii): `"home-manager, unstable-20.03pre196836.1c40ee6fc44"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/python-modules/pycuda/default.nix
+++ b/pkgs/development/python-modules/pycuda/default.nix
@@ -13,7 +13,6 @@
 , python
 , mkDerivation
 , stdenv
-, isPy3k
 }:
 let
   compyte = import ./compyte.nix {
@@ -29,11 +28,12 @@ buildPythonPackage rec {
     sha256 = "ada56ce98a41f9f95fe18809f38afbae473a5c62d346cfa126a2d5477f24cc8a";
   };
 
-  preConfigure = ''
+  preConfigure = with stdenv.lib.versions; ''
     ${python.interpreter} configure.py --boost-inc-dir=${boost.dev}/include \
                           --boost-lib-dir=${boost}/lib \
                           --no-use-shipped-boost \
-                          --boost-python-libname=boost_python${stdenv.lib.optionalString isPy3k "3"}
+                          --boost-python-libname=boost_python${major python.version}${minor python.version} \
+                          --cuda-root=${cudatoolkit}
   '';
 
   postInstall = ''

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -3268,7 +3268,7 @@ in {
   pycares = callPackage ../development/python-modules/pycares { };
 
   pycuda = callPackage ../development/python-modules/pycuda {
-    cudatoolkit = pkgs.cudatoolkit_7_5;
+    cudatoolkit = pkgs.cudatoolkit;
     inherit (pkgs.stdenv) mkDerivation;
   };
 

